02/styles/global.css

```diff
@@ -14,13 +16,38 @@
 }
 
 .a {
-    border-color: #ffffba
+    border-color: #ffffba;
+    grid-area: header;
+
 }
 
 .b {
     border-color: #ffdfba;
+    grid-area: section;
 }
 
 .c {
     border-color: #baffc9;
+    order: 1;
+    grid-area: main;
+}
+
+body {
+    display: grid;
+    grid-template-areas:
+        "header"
+        "section"
+        "main";
+}
+
+@media (min-width: 600px) {
+    body {
+        display: grid;
+        grid-template-areas:
+            "header header"
+            "main section";
+        grid-template-columns: 50% 50%;
```

Comment:
Jeśli chodzi o wysokość to lepiej użyć 50vh bo przy 50% bierzemy pod uwagę rodzica i jeśli on nie zajmuje 100% wysokości okna to te 50% to jest nie wiele :)